### DISCUSSION TITLE
Should different submarine classes (other than deep diver) give benefits?

### DISCUSSION THREAD

**Discussion by RadCatish** [FakeFishGames/Barotrauma#4899] [Ideas and suggestions]
Not my idea, but on discord it was suggested that deep diver shouldn't be the only submarine class to come with benefits. I personally agree with that, it would be interesting if subs had strengths and weaknesses

![image](https://user-images.githubusercontent.com/77641131/105763343-96ad6a80-5f4d-11eb-9d09-1c3a014516a5.png)

**Comment by 3e849f2e5c** [comment]
I believe something is in the plans for this already, submarine classes were introduced in New Frontiers update and their original purpose was that you could only buy one submarine of each class, of course the testers made it clear that being locked to only 4 submarines was dumb so the cap was removed and classes were left as a cosmetic only and are now being slowly re-purposed.

**Comment by Xeavas** [comment]
Tbh, I'd still rather it be flavour text rather than giving definite stat advantages.

But as a compromise, allow all subs to be upgraded equally, but upgrades are 20% cheaper if its class related, and subs will start with some basic upgrades relating to said class with the exception of the starting sub.

This doesn't factor in transport classes though, that's another can of worms.

**Comment by ubertpendragon** [comment]
The problem with sub classes is that their "flavor" doesn't actually translate into gameplay

Transport subs are somewhat meaningless, since transport missions only require you to carry two(!) crates, as well as not being a "real" mission (going from point A to point B), so every submarine is a transport submarine. Both Berillia and R29 have that "big cargo" flavor, but said cargo room has no purpose due to the nature of transport mission [(see: proposed transport overhaul)
](https://cdn.discordapp.com/attachments/220881718666723328/773917044185759745/Barotransport.png)
Any advanced submarine also has a combination of both scouting and attack capabilities, via drones, special weaponry and other systems, so all sub classes are effectively combined into all-pupose vessels. Scouts can be small: but the game adapts the cave size to your sub. Attackers can be weaponized, but every sub is better with more weapons.

The "Deep Diver" class exists, but the game has never really communicated what "deep diving" is, what it means in terms of navigation, enemy encounter, exporation. _What is "deep" in Baro?_ It fails in both flavor and purpose.

I don't think classes will come into their own with passive bonuses: if the game doesn't push for sub differenciation via gameplay mechanics, **you'll end up with all purpose subs and people picking the tag with the best passive.** As the game expands and we get more mission/level/enemy/hazard variety. But these need to be designed with classes in mind. Here's an old relevant pic of mine:

![Baroclasses](https://user-images.githubusercontent.com/73229309/105860444-de83ce80-5fed-11eb-8cb6-543080f48f89.png)
